src/CollectionsAndGenerics/DictionaryOperations.cs

```diff
@@ -0,0 +1,137 @@
+﻿// <copyright file="DictionaryOperations.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace CollectionsAndGenerics
+{
+    /// <summary>
+    /// DictionaryOperations class - Adds, removes and displays grades of students
+    /// </summary>
+    /// <typeparam name="TKey">Key Generic type parameter</typeparam>
+    /// <typeparam name="TValue">Value Generic type parameter</typeparam>
+    public class DictionaryOperations<TKey, TValue>
+        where TKey : notnull
+    {
+        /// <summary>
+        /// Performs Dictionary Operations
+        /// </summary>
+        private enum OperationsByDictionary
+        {
+            AddStudentGrades = 1,
+            RemoveStudentGrade = 2,
+            Display = 3,
+            Exit = 4,
+        }
+
+        /// <summary>
+        /// Gets or Sets GradeOfStudents
+        /// </summary>
+        /// <value>Name of the student as key and their grades as values</value>
+        public Dictionary<TKey, TValue> GradeOfStudents { get; set; } = new Dictionary<TKey, TValue>(5);
+
+        /// <summary>
+        /// AddStudentGradesInDictionary - Adds student grades to the queue
+        /// </summary>
+        public virtual void AddStudentGradesInDictionary()
+        {
+            if (this.GradeOfStudents.Count == 5)
+            {
+                Console.WriteLine("\nDictionary is Full!");
+                return;
+            }
+
+            Console.WriteLine("\nSPECIFY STUDENT NAME AND GRADE");
+            string studentName = Program.GetValidName<string>("Name of the Student", "^[a-zA-Z]+($|( +([a-zA-Z])*)*$)");
+            if (this.GradeOfStudents.ContainsKey((TKey)(object)studentName))
+            {
+                Console.WriteLine($"\nStudent named {studentName} is already available!");
+                return;
+            }
+
+            double grade = ValidateInput.Program.GetValidInput<double>($"Grade of the {studentName}");
+            this.GradeOfStudents.Add((TKey)(object)studentName, (TValue)(object)grade);
+        }
+
+        /// <summary>
+        /// IsEmpty - to check if a dictionary is Empty
+        /// </summary>
+        /// <returns>Returns true if empty else false</returns>
+        public bool IsEmpty()
+        {
+            return this.GradeOfStudents.Count == 0 ? true : false;
```

Comment:
```suggestion
            return this.GradeOfStudents.Any() ? false : true;
```

Applies to similar scenarios

Reply:
Modified all IsEmpty() function